### PR TITLE
BUG Avoid IndexError on writing empty (row) MI df to excel (GH19543)

### LINKED ISSUE #19543: BUG: to_excel to throw IndexError for dataframe with empty (row) MultiIndex
```python
mi = pd.MultiIndex.from_arrays([
        np.array(['a', 'a', 'b', 'b']), 
        np.array(['1', '2', '2', '3']), 
        np.array(['alpha', 'beta', 'alpha', 'beta']),
    ], names=['one', 'two', 'three'])
df = pd.DataFrame(np.random.rand(4, 3), index=mi)
df2 = df.loc[('b', '1', slice(None)), :]
print(df2.index)
```
Output:
```
MultiIndex(levels=[['a', 'b'], ['1', '2', '3'], ['alpha', 'beta']],
           labels=[[], [], []],
           names=['one', 'two', 'three'])
```
The next line causes an `IndexError`:
```python
df2.to_excel('test.xlsx')
```
Notice that the 'empty' `MultiIndex` does not give any indication of its presence when the entire dataframe is printed, looking like an ordinary empty dataframe instead. 
```
Empty DataFrame
Columns: [0, 1, 2]
Index: []
```
#### Problem description

I'm not quite sure which behaviour is the problematic one here: the fact that the first block of code creates an 'empty' `MultiIndex` that has values in `levels`, or that `to_excel` doesn't write a `DataFrame` with such an index to file properly. The 'empty' `MultiIndex` behaviour is a little bit anomalous by itself, because a similar block of code (below) will cause a `KeyError`. The `KeyError` is 'suppressed' by the fact that there is a third level in the `MultiIndex`.

```python
mi = pd.MultiIndex.from_arrays([
        np.array(['a', 'a', 'b', 'b']), 
        np.array(['1', '2', '2', '3']), 
    ], names=['one', 'two'])
df = pd.DataFrame(np.random.rand(4, 3), index=mi)
df2 = df.loc[('b', '1'), :]
```

I initially ran into this behaviour (the IndexError) when using `xs`, but since `xs` seems to be headed for the deprecation guillotine, I figured I'd rewrite this with `loc` instead.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-43-Microsoft
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.22.0
pytest: None
pip: 9.0.1
setuptools: 27.2.0
Cython: None
numpy: 1.14.0
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: None
patsy: None
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: None
tables: 3.4.2
numexpr: 2.6.4
feather: None
matplotlib: None
openpyxl: 2.4.9
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 1.0b10
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by wf-r**
- [x] closes #19543 
- [x] Tests added and passed
- [x] All code checks passed
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/io/excel/test_writers.py [hunk 1]
```diff
@@ -838,6 +838,26 @@ def test_to_excel_multiindex_no_write_index(self, path):
         # Test that it is the same as the initial frame.
         tm.assert_frame_equal(frame1, frame3)
 
+    def test_to_excel_empty_multiindex(self, path):
+        # Test writing and re-reading an empty MI. GH 19543.
```
Comment (phofl) [inline pandas/tests/io/excel/test_writers.py:842]: You can remove the comments, the gh reference is sufficient

File: pandas/tests/io/excel/test_writers.py [hunk 2]
```diff
@@ -838,6 +838,26 @@ def test_to_excel_multiindex_no_write_index(self, path):
...
+
+        # Initial non-MI frame.
+        frame1 = DataFrame([[], [], []]).T
+
+        # Add a MI.
+        frame2 = frame1.set_index([0, 1])
```
Comment (phofl) [inline pandas/tests/io/excel/test_writers.py:848]: Could you define both of them explicitly without using any other functions like .T or set_index?

Comment (phofl) [reply pandas/tests/io/excel/test_writers.py:848]: Also please call them df, result and expected

File: doc/source/whatsnew/v1.5.0.rst [hunk 3]
```diff
@@ -832,6 +832,7 @@ I/O
 - :meth:`to_html` now excludes the ``border`` attribute from ``<table>`` elements when ``border`` keyword is set to ``False``.
 - Bug in :func:`read_sas` returned ``None`` rather than an empty DataFrame for SAS7BDAT files with zero rows (:issue:`18198`)
 - Bug in :class:`StataWriter` where value labels were always written with default encoding (:issue:`46750`)
+- Bug in :func:`to_excel` when writing an empty dataframe with row :class:`MultiIndex` (:issue:`19543`)
```
Comment (phofl) [inline doc/source/whatsnew/v1.5.0.rst:835]: This is not a func, rather a method, :meth:`DataFrame.to_excel` and you can remove *row*

**Review by phofl — CHANGES_REQUESTED** [review]

**Comment by wf-r** [comment]
@github-actions pre-commit

**Comment by wf-r** [comment]
@phofl Thanks for your feedback, I think you can take another look.

File: pandas/tests/io/excel/test_writers.py [hunk 4]
```diff
@@ -838,6 +838,28 @@ def test_to_excel_multiindex_no_write_index(self, path):
         # Test that it is the same as the initial frame.
         tm.assert_frame_equal(frame1, frame3)
 
+    def test_to_excel_empty_multiindex(self, path):
+        # GH 19543.
+
+        # Initial non-MI frame.
```
Comment (phofl) [inline pandas/tests/io/excel/test_writers.py:844]: Please remove all comments

File: pandas/tests/io/excel/test_writers.py [hunk 5]
```diff
@@ -838,6 +838,18 @@ def test_to_excel_multiindex_no_write_index(self, path):
         # Test that it is the same as the initial frame.
         tm.assert_frame_equal(frame1, frame3)
 
+    def test_to_excel_empty_multiindex(self, path):
```
Comment (phofl) [inline pandas/tests/io/excel/test_writers.py:841]: The gh reference has to stay (see my first comment about this)

Comment (wf-r) [reply pandas/tests/io/excel/test_writers.py:841]: 😂 done (I do not think the test errors were caused by my changes)

File: doc/source/whatsnew/v1.5.0.rst [hunk 6]
```diff
@@ -872,6 +872,7 @@ I/O
 - Bug in :func:`read_sas` returned ``None`` rather than an empty DataFrame for SAS7BDAT files with zero rows (:issue:`18198`)
 - Bug in :class:`StataWriter` where value labels were always written with default encoding (:issue:`46750`)
 - Bug in :class:`StataWriterUTF8` where some valid characters were removed from variable names (:issue:`47276`)
+- Bug in :meth:`to_excel` when writing an empty dataframe with :class:`MultiIndex` (:issue:`19543`)
```
Comment (phofl) [inline doc/source/whatsnew/v1.5.0.rst:875]: DataFrame.to_excel please

Please don't resolve my remarks when I haven't checked them over. Normally the author of the remarks clicks resolve

Comment (wf-r) [reply doc/source/whatsnew/v1.5.0.rst]: 🙈 done

**Review by phofl — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @wf-r

**Pull request merged**